eth/stagedsync/stage_execute.go

```diff
@@ -204,11 +204,11 @@ func gatherNoPruneReceipts(receipts *types.Receipts, chainCfg *chain.Config) boo
 	cr := types.Receipts{}
 	for _, r := range *receipts {
 		toStore := false
-		if chainCfg.DepositContract != nil && *chainCfg.DepositContract == r.ContractAddress {
+		if chainCfg.DepositContract == r.ContractAddress {
```

Comment:
Why remove `DepositContract != nil` checks? `DepositContract` is `nil` for Polygon.

Reply:
Changed from pointer to value, under the assumption that it is never ZERO address

Reply:
OK. The underlying assumption is that no EVM logs can be emitted from address ZERO, which is probably reasonable.